edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/NestedDeviceScopeApiClient.cs

```diff
@@ -87,10 +87,12 @@ public class NestedDeviceScopeApiClient : IDeviceScopeApiClient
         public Task<ScopeResult> GetNextAsync(string continuationToken) =>
             this.GetIdentitiesInTargetScopeWithRetry(this.GetNestedScopeServiceUri(), Option.Some(continuationToken));
 
-        public Task<ScopeResult> GetIdentityAsync(string targetDeviceId, string targetModuleId)
+        public Task<ScopeResult> GetIdentityAsync(string deviceId, string moduleId) => throw new NotImplementedException("Use GetIdentityOnBehalfOfAsync() instead");
+
+        public Task<ScopeResult> GetIdentityOnBehalfOfAsync(string targetDeviceId, Option<string> targetModuleId, string onBehalfOfDevice)
```

Comment:
check onBehalfOfDevice not null or empty?  And onBehalfOfDevice is the auth chain, right? will it contain edgeHub as part of the value? if so, using "Device" wording may be confusing.

Reply:
onBehalfOfDevice is the auth chain, right? will it contain edgeHub as part of the value? if so, using "Device" wording may be confusing.

Reply:
no, the onBehalfOfDevice is the ID of the first Edge that received the leaf connection, so it's just a DeviceId.